src/component/feature/FeatureStrategy/FeatureStrategyConstraints2/FeatureStrategyConstraints2.tsx

```diff
@@ -0,0 +1,110 @@
+import { IFeatureStrategy, IConstraint } from 'interfaces/strategy';
+import React, { useState } from 'react';
+import { ConstraintAccordion } from 'component/common/ConstraintAccordion/ConstraintAccordion';
+import produce from 'immer';
+import useUnleashContext from 'hooks/api/getters/useUnleashContext/useUnleashContext';
+import PermissionButton from 'component/common/PermissionButton/PermissionButton';
+import {
+    CREATE_FEATURE_STRATEGY,
+    UPDATE_FEATURE_STRATEGY,
+} from 'component/providers/AccessProvider/permissions';
+import { FeatureStrategyFormMode } from 'component/feature/FeatureStrategy/FeatureStrategyForm/FeatureStrategyForm';
+import { createEmptyConstraint } from 'component/feature/FeatureStrategy/FeatureStrategyConstraints2/createEmptyConstraint';
+
+interface IFeatureStrategyConstraints2Props {
+    mode: FeatureStrategyFormMode;
+    projectId: string;
+    environmentId: string;
+    strategy: Partial<IFeatureStrategy>;
+    setStrategy: React.Dispatch<
+        React.SetStateAction<Partial<IFeatureStrategy>>
+    >;
+}
+
+export const FeatureStrategyConstraints2 = ({
+    mode,
+    projectId,
+    environmentId,
+    strategy,
+    setStrategy,
+}: IFeatureStrategyConstraints2Props) => {
+    const [editIndexes, setEditIndexes] = useState<number[]>([]);
+    const { context } = useUnleashContext();
+
+    const stopEditingAll = () => {
+        setEditIndexes([]);
+    };
+
+    const startEditingIndex = (index: number) => {
+        setEditIndexes(prev => [...prev, index]);
+    };
+
+    const stopEditingIndex = (index: number) => {
+        setEditIndexes(prev => prev.filter(editIndex => editIndex !== index));
+    };
+
+    const isEditingIndex = (index: number) => {
+        return editIndexes.includes(index);
+    };
+
+    const onAddConstraint = () => {
+        if (mode === 'create') {
```

Comment:
This is almost accurate. If you add a new strategy in edit mode, this should also start in "Editing mode".

Reply:
In addition, if you hit cancel on a new constraint (either in create or edit mode) without first saving the draft, it should remove the constraint from the list. This was the reasoning behind the metadata in the previous implementation.